libeos-updater-util/flatpak.c

```diff
@@ -1457,6 +1457,651 @@ euu_filter_for_existing_flatpak_ref_actions (GHashTable *ref_actions  /* (elemen
+            return FALSE;
+        }
+    }
+
+  return TRUE;
+}
+
+static GPtrArray *
+list_installed_related_refs_for_remote (FlatpakInstallation  *installation,
+                                        const gchar          *remote_name,
+                                        FlatpakRef           *source_ref,
+                                        const gchar          *formatted_ref,
+                                        GCancellable         *cancellable,
+                                        GError              **error)
+{
+  g_autoptr(GPtrArray) installed_related_refs = NULL;
+  g_autoptr(FlatpakRemoteRef) remote_runtime_ref = NULL;
+  FlatpakRef *remote_runtime_ref_as_ref;
+  g_autoptr(FlatpakInstalledRef) installed_ref = NULL;
+  g_autoptr(GError) local_error = NULL;
+
+  installed_related_refs =
+    flatpak_installation_list_installed_related_refs_sync (installation,
+                                                           remote_name,
+                                                           formatted_ref,
+                                                           cancellable,
+                                                           error);
+
+  /* Only error that can occurr here is FLATPAK_ERROR_NOT_INSTALLED
+   * and it is the caller's responsibility to check if the source
+   * flatpak is installed. All errors are thus fatal. */
+  if (installed_related_refs == NULL)
+    return NULL;
+
+  /* Try and fetch the remote ref for the runtime, but if there isn't one
+   * then it is not an error */
+  if (!fetch_remote_runtime_ref (installation,
+                                 remote_name,
+                                 source_ref,
+                                 cancellable,
+                                 &remote_runtime_ref,
+                                 error))
+    return NULL;
+
+  if (remote_runtime_ref == NULL)
+    return g_steal_pointer (&installed_related_refs);
+
+  remote_runtime_ref_as_ref = FLATPAK_REF (remote_runtime_ref);
+  installed_ref = flatpak_installation_get_installed_ref (installation,
+                                                          flatpak_ref_get_kind (remote_runtime_ref_as_ref),
+                                                          flatpak_ref_get_name (remote_runtime_ref_as_ref),
+                                                          flatpak_ref_get_arch (remote_runtime_ref_as_ref),
+                                                          flatpak_ref_get_branch (remote_runtime_ref_as_ref),
+                                                          cancellable,
+                                                          &local_error);
+
+  if (installed_ref == NULL)
+    {
+      if (!g_error_matches (local_error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_INSTALLED))
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return NULL;
+        }
+    }
+
+  g_ptr_array_add (installed_related_refs, g_steal_pointer (&installed_ref));
```

Comment:
Should this line be skipped if `installed_ref == NULL`?

Reply:
It should be, yes